tests/app/main/views/test_suppliers.py

```diff
@@ -1268,53 +1300,130 @@ def test_none_shown_if_user_not_ccs_admin(self, s3, get_signed_url, data_api_cli
         res = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
         assert res.status_code == 200
 
-        data = res.get_data(as_text=True)
+        document = html.fromstring(res.get_data(as_text=True))
 
-        assert 'Accept and continue' not in data
-        assert 'Put on hold and continue' not in data
+        assert len(document.xpath("//input[@type='submit'][@value='Accept and continue']")) == 0
```

Comment:
I find these very specific tests for where something *isn't* to be less convincing than the previous "text is nowhere in the page" tests.  I agree that for checking that something *is* rendered on the page then these xpath specific checks are better, but for asserting things *not* being there I think it's better to cast a wider net. (Because designs and HTML structure can change and leave the specific test passing even though the unwanted text now appears on the page in a slightly different element).